ghc-lib-gen/src/Ghclibgen.hs

```diff
@@ -1002,7 +1052,7 @@ generatePrerequisites ghcFlavor = do
           , "--build-root=ghc-lib"
         ] ++
         (if ghcFlavor >= Ghc901 then ["--bignum=native"] else ["--integer-simple"]) ++
-        ghcLibParserExtraFiles ghcFlavor ++ map (dataDir </>) dataFiles
+        ghcLibParserExtraFiles ghcFlavor [] ++ map (dataDir </>) (dataFiles ghcFlavor)
```

Comment:
This seems a bit backwards. Instead of defining source and target, you only define a function that applies a substittion to the source and then apply it to an empty substitution in the places where you need the source. Can we extract the source lists to a top-level definition and then use that here directly? I think that’d be clearer than what we do now.

Reply:
ok